File: QE-Java Selenium/Activity15.java
package demos;
import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.*;
import java.time.Duration;
import java.util.Random;

public class Activity15 {
    
    static String randText(int n) {
        String chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
        StringBuilder sb = new StringBuilder();
        Random r = new Random();
        for (int i = 0; i < n; i++) {
            sb.append(chars.charAt(r.nextInt(chars.length())));
        }
        return sb.toString();
    }

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(15));

        driver.get("https://training-support.net/webelements/dynamic-attributes");
        System.out.println("Page Title: " + driver.getTitle());

        //String name = randText(6);
        String name = "Sasi";
        String email = randText(6) + "@test.com";
        String password = randText(6) + "@1";
        String details = randText(10);
        String dateValue = "2026-02-01"; 

        wait.until(ExpectedConditions.presenceOfElementLocated(
                By.xpath("//input[contains(@id,'full-name')]")
        )).sendKeys(name);

        driver.findElement(By.xpath("//input[contains(@id,'email')]")).sendKeys(email);
        driver.findElement(By.xpath("//*[@id='signInForm']/input[3]")).sendKeys(password);
        driver.findElement(By.xpath("//textarea[contains(@id,'additional-details')]")).sendKeys(details);

        WebElement dateField = driver.findElement(By.xpath("//input[@type='date']"));
        ((JavascriptExecutor) driver).executeScript(
                "arguments[0].value = arguments[1];", dateField, dateValue
        );

        driver.findElement(By.xpath("//*[@id='signInForm']/button")).click();

        WebElement success = wait.until(ExpectedConditions.visibilityOfElementLocated(
                By.xpath("//*[@id='action-confirmation']")
        ));
        System.out.println("Success Message: " + success.getText());

        driver.quit();
    }
}

File: QE-Python Selenium/Activity15.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import random, string

def rand(n=6):
    return ''.join(random.choices(string.ascii_letters, k=n))

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 15)

driver.get("https://training-support.net/webelements/dynamic-attributes")
print("Page Title:", driver.title)

name = rand()
email = rand() + "@test.com"
password = rand(6) + "@1"
details = rand(10)
date_value = "2026-02-01"   # YYYY-MM-DD (mandatory format)

wait.until(EC.presence_of_element_located(
    (By.XPATH, "//input[contains(@id,'full-name')]")
)).send_keys(name)

driver.find_element(By.XPATH, "//input[contains(@id,'email')]").send_keys(email)
driver.find_element(By.XPATH, "//*[@id='signInForm']/input[3]").send_keys(password)
driver.find_element(By.XPATH, "//textarea[contains(@id,'additional-details')]").send_keys(details)


date_field = driver.find_element(By.XPATH, "//input[@type='date']")
driver.execute_script("arguments[0].value = arguments[1];", date_field, date_value)

driver.find_element(By.XPATH, "//*[@id='signInForm']/button").click()

success = wait.until(EC.visibility_of_element_located(
    (By.XPATH, "//*[@id='action-confirmation']")
))
print("Success Message:", success.text)

driver.quit()
